interface.py

```diff
@@ -0,0 +1,621 @@
+        ]
+        self.castling_button = []
+        self.castling_button.append(
+            Button(
+                self.castling_box,
+                image=image[0],
+                bg=self.main.color_palette[7],
+                activebackground=self.main.color_palette[8],
+                command=partial(self.display_castling, 0),
+            )
+        )
+        self.castling_button.append(
+            Button(
+                self.castling_box,
+                image=image[1],
+                bg=self.main.color_palette[7],
+                activebackground=self.main.color_palette[8],
+                command=partial(self.display_castling, 1),
+            )
+        )
+        self.castling_button.append(
+            Button(
+                self.castling_box,
+                image=image[2],
+                bg=self.main.color_palette[7],
+                activebackground=self.main.color_palette[8],
+                command=partial(self.display_castling, 2),
+            )
+        )
+        self.castling_button.append(
+            Button(
+                self.castling_box,
+                image=image[3],
+                bg=self.main.color_palette[7],
+                activebackground=self.main.color_palette[8],
+                command=partial(self.display_castling, 3),
+            )
+        )
+
+        for i in range(4):
+            self.castling_button[i].image = image[i]
+
+        self.castling_button[1].place(relx=0, rely=0.5, relwidth=0.5, relheight=0.5)
+        self.castling_button[3].place(relx=0, rely=0, relwidth=0.5, relheight=0.5)
+        self.castling_button[0].place(relx=0.5, rely=0.5, relwidth=0.5, relheight=0.5)
+        self.castling_button[2].place(relx=0.5, rely=0, relwidth=0.5, relheight=0.5)
+
+        size = self.get_size(option_width / 2)
+        image = Image.open(
+            home_path + "\\img\\" + str(size) + pieces_name + pieces_path["K"] + ".png"
+        )
+        image = ImageTk.PhotoImage(image)
+        self.player_w = Button(
+            self.player_box,
+            bg=self.main.color_palette[3],
+            activebackground=self.main.color_palette[8],
+            command=partial(self.display_player, "w"),
+            image=image,
+        )
+        self.player_w.image = image
+        image = Image.open(
+            home_path + "\\img\\" + str(size) + pieces_name + pieces_path["k"] + ".png"
+        )
+        image = ImageTk.PhotoImage(image)
+        self.player_b = Button(
+            self.player_box,
+            bg=self.main.color_palette[7],
+            activebackground=self.main.color_palette[8],
+            command=partial(self.display_player, "b"),
+            image=image,
+        )
+        self.player_b.image = image
+
+        self.player_w.place(relx=0, rely=0, relwidth=0.5, relheight=1)
+        self.player_b.place(relx=0.5, rely=0, relwidth=0.5, relheight=1)
+
+    def select(self, column, row, piece):
+        global selected_piece

```

Comment:
I am against storing state in global variables. What other options of storing state do you have?

Reply:
I was thinking about storing them in the Main class